File: apps/guzo-api/src/main/java/et/guzo/web/controller/BranchStaffController.java
package et.guzo.web.controller;

import et.guzo.common.ApiResponse;
import et.guzo.domain.entity.BranchStaff;
import et.guzo.security.RoleChecker;
import et.guzo.security.SecurityUtil;
import et.guzo.service.BranchStaffService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/branch-staff")
@RequiredArgsConstructor
public class BranchStaffController {

    private final BranchStaffService branchStaffService;

    @PostMapping
    public ApiResponse<BranchStaff> assign(@RequestBody Map<String, String> body) {
        RoleChecker.requireAdmin(SecurityUtil.requireUser());
        return ApiResponse.ok(branchStaffService.assign(body.get("userId"), body.get("branchId")), "Staff assigned");
    }

    @GetMapping("/branch/{branchId}")
    public ApiResponse<List<BranchStaff>> list(@PathVariable String branchId) {
        RoleChecker.requireAdmin(SecurityUtil.requireUser());
        return ApiResponse.ok(branchStaffService.listByBranch(branchId));
    }

    @GetMapping("/me")
    public ApiResponse<List<Map<String, Object>>> me() {
        return ApiResponse.ok(branchStaffService.myBranches(SecurityUtil.requireUser().getId()));
    }
}

File: apps/server/src/modules/branches/branch-staff.controller.ts
import type { Request, Response } from 'express';
import { asyncHandler } from '../../utils/asyncHandler.js';
import { ok } from '../../utils/ApiResponse.js';
import { ApiError } from '../../utils/ApiError.js';
import { branchStaffService } from './branch-staff.service.js';

export const branchStaffController = {
  assign: asyncHandler(async (req: Request, res: Response) => {
    const { userId, branchId } = req.body ?? {};
    if (!userId || !branchId) throw ApiError.badRequest('userId and branchId are required');
    const data = await branchStaffService.assign(userId, branchId);
    return ok(res, data, 'Staff assigned');
  }),

  listByBranch: asyncHandler(async (req: Request, res: Response) => {
    const data = await branchStaffService.listByBranch(req.params.branchId);
    return ok(res, data, 'Branch staff loaded');
  }),

  me: asyncHandler(async (req: Request, res: Response) => {
    const data = await branchStaffService.myBranches(req.user!.id);
    return ok(res, data, 'My branches');
  }),
};
